modules/private.php

```diff
@@ -0,0 +1,271 @@
+	 */
+	static function privatize_blog_mask_blog_name( $value, $what ) {
+		if ( in_array( $what, array( 'name', 'title' ), true ) ) {
+			$value = __( 'Protected Blog', 'jetpack' );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Filters new comments so that users can't comment on private blogs
+	 *
+	 * @param array $comment Documented in wp-includes/comment.php.
+	 */
+	static function privatize_blog_comments( $comment ) {
+		Jetpack_Private::privatize_blog( null );
+		return $comment;
+	}
+
+	/**
+	 * Extend the 'Site Visibility' privacy options to also include a private option
+	 **/
+	static function privatize_blog_priv_selector() {
+		?>
+		<style>
+			.jetpack-private__setting-disabled {
+				font-weight: bold;
+				padding: 10px;
+			}
+			.option-site-visibility fieldset {
+				display: none;
+			}
+		</style>
+		</fieldset>
+		<div class="jetpack-private__setting-disabled highlight">
+			<?php printf( __( 'This setting is ignored because you <a href="%s">made your site private</a>', 'jetpack' ), admin_url( 'admin.php?page=jetpack' ) . '#/traffic?term=private' ); ?>
+		</div>
+		<fieldset>
+		<?php
+	}
+
+	/**
+	 * Don't let search engines index private sites
+	 * or sites not deemed publicly available, like deleted, archived, spam.
+	 *
+	 * @param string $output Robots.txt output.
+	 */
+	static function private_robots_txt( $output ) {
+		$output  = "User-agent: *\n"; // Purposefully overriding current output; we only want these rules.
+		$output .= "Disallow: /\n";
+		return $output;
+	}
+
+	/**
+	 * Output the meta tag that tells Pinterest not to allow users to pin
+	 * content from this page.
+	 * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+	 */
+	static function private_no_pinning() {
+		echo '<meta name="pinterest" content="nopin" />';
+	}
+
+	/**
+	 * Prevents ajax requests on private blogs for users who don't have permissions
+	 *
+	 * @param string    $action The Ajax nonce action.
+	 * @param false|int $result The result of the nonce check.
+	 */
+	static function private_blog_ajax_nonce_check( $action, $result ) {
+		global $current_user, $wpdb;
+
+		if ( is_super_admin() ) {
+			return;
+		}
+
+		if ( 1 !== $result && 2 !== $result ) {
+			return;
+		}
+
+		if ( 'find-posts' !== $action && 'internal-linking' !== $action ) {
+			return;
+		}
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {
+			return;
+		}
+
+		if ( ! Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) {
+			wp_die( -1 );
+		}
+	}
+
+	/**
+	 * Disables WordPress Rest API for external requests
+	 */
+	static function disable_rest_api() {
+		global $pagenow, $current_user, $wpdb;
+
+		if ( $current_user && ( is_super_admin() || Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) ) {
+			return;
+		}
+
+		die( __( 'This site is private.', 'jetpack' ) );
```

Comment:
* Should this `403`?
* Should it call `wp_die` instead of `die`?

Reply:
Good idea 👍